presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -220,6 +230,145 @@ public void testYield()
+                        revokeMemory(buildOperator);
+                        revoked.set(i, true);
+                    }
+                }
+            }
+            getFutureValue(lookupSourceProvider).close();
+            assertEquals(revoked, whenSpill.stream().map(WhenSpill.DURING_BUILD::equals).collect(toImmutableList()), "Some operators not spilled before LookupSource built");
+
+            for (int i = 0; i < buildOperatorCount; i++) {
+                if (whenSpill.get(i) == WhenSpill.AFTER_BUILD) {
+                    revokeMemory(buildSideSetup.buildOperators.get(i));
+                }
+            }
+
+            for (Driver buildDriver : buildDrivers) {
+                runDriver(executor, buildDriver);
+            }
+
+            // process join
+            Iterator<Page> input = probePages.build().iterator();
+            List<Page> actualPages = new ArrayList<>();
+            actualPages.addAll(toPagesPartial(joinOperator, singletonIterator(input.next())));
+
+            for (int i = 0; i < buildOperatorCount; i++) {
+                if (whenSpill.get(i) == WhenSpill.DURING_USAGE) {
+                    checkState(input.hasNext(), "spill requested DURING_USAGE, but input is kind of exhausted");
+                    triggerMemoryRevokingAndWait(buildSideSetup.buildOperators.get(i));
+                }
+            }
+
+            actualPages.addAll(toPages(joinOperator, input));
+
+            // expected
+            MaterializedResult expected = MaterializedResult.resultBuilder(taskContext.getSession(), concat(probePages.getTypesWithoutHash(), buildPages.getTypesWithoutHash()))
+                    .row("20", 123_020L, "20", 200L)
+                    .row("21", 123_021L, "21", 201L)
+                    .row("22", 123_022L, "22", 202L)
+                    .row("23", 123_023L, "23", 203L)
+                    .row("30", 123_000L, "30", 300L)
+                    .row("31", 123_001L, "31", 301L)
+                    .row("32", 123_002L, "32", 302L)
+                    .row("33", 123_003L, "33", 303L)
+                    .build();
+
+            List<Type> types = joinOperator.getTypes();
+            if (probePages.getHashChannel().isPresent()) {
+                List<Integer> hashChannels = ImmutableList.of(probePages.getHashChannel().get());
+                actualPages = dropChannel(actualPages, hashChannels);
+                types = without(types, hashChannels);
+            }
+
+            MaterializedResult actual = OperatorAssertion.toMaterializedResult(joinOperator.getOperatorContext().getSession(), types, actualPages);
+            assertEqualsIgnoreOrder(actual.getMaterializedRows(), expected.getMaterializedRows());
+        }
+    }
+
+    private enum WhenSpill
+    {
+        DURING_BUILD, AFTER_BUILD, DURING_USAGE, NEVER
+    }
+
+    @DataProvider
+    public Object[][] testInnerJoinWithSpillDataProvider()
```

Comment:
This (and the enum) should go above `testInnerJoinWithSpill`... also, you don't need `DataProvider` in the name